src/backend/distributed/planner/multi_physical_planner.c

```diff
@@ -2821,6 +2825,42 @@ SqlTaskList(Job *job)
 }
 
 
+/*
+ * RelabelTypeMutator walks over the tree and converts RelabelType's
+ * into CollationExpr's. With that, we will be able to pushdown COLLATE's.
+ */
+static Node *
+RelabelTypeMutator(Node *originalNode)
+{
+	if (originalNode == NULL)
+	{
+		return NULL;
+	}
+
+	if (IsA(originalNode, RelabelType))
+	{
+		RelabelType *originalRelabelType = (RelabelType *) originalNode;
+		if (OidIsValid(originalRelabelType->resultcollid))
+		{
+			CollateExpr *newCollateExpr = makeNode(CollateExpr);
+			newCollateExpr->arg = originalRelabelType->arg;
```

Comment:
One slight concern I have is that there might be relabel types that change both the type (modifier) and the collation, and we're throwing away the information about the former. However, I don't know whether this actually happens in practice, but it's worth paying attention to during testing.